main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -55,6 +55,13 @@ object Unification {
     Result.Ok(Substitution.singleton(x, tpe))
   }
 
+  def canSubstituteKind(kind1: Kind, kind2: Kind): Boolean = (kind1, kind2) match {
+    case _ if (kind1 == kind2) => true
+    case (Kind.Unbound, _) => true
```

Comment:
Is there a particular reason this is not symmetric?

Reply:
Do you mean "why there is not a case `(_, Kind.Unbound)`? See also above comment.

My understanding is that if we allow substitution from more "specific" to less "specific" kinds we could end up with the following situation:

```
tvarR :: Kind.Record
tvarS :: Kind.Schema
tvarU :: Kind.Unbound

let substRU = unify(tvarR, tvarU) = subst(tvarR, tvarU)
let substSU = unify(tvarS, tvarU) = subst(tvarS, tvarU)

let badDef = "def bad[a](r: { x: Int | r }, s: #{ Y(Int) | r }): Int = ..."

merge(substRU, substSU).apply(badDef)
Success! (but should fail because of mismatched kinds)
```

Reply:
Hmmm. Let us see it works out. I think this will be solved once we eventually make `KindUnbound` into a `Kind.Var` because then you substitute `Kind.Var` for whatever kind is the right one and the `canSubstitute` function goes away.